aieng-ui/backend/app/runtime_registry/cae.py

```
@@ -2037,10 +2089,31 @@ def _tool_cae_run_solver(inp: dict[str, Any], _ctx: dict[str, Any]) -> dict[str,
         solver: str = inp.get("solver") or "CalculiX"
         load_case_id: str = inp.get("loadCaseId") or inp.get("load_case_id") or "load_case_001"
         input_deck_path_str: str | None = inp.get("inputDeckPath") or inp.get("input_deck_path")
+        # The run the deck belongs to. `run_id` used to default to "run_001"
+        # regardless of the deck given, so solving
+        # `simulation/runs/run_002/solver_input.inp` wrote its FRD, log and
+        # `solver_run.json` OVER run_001 — destroying the baseline the whole
+        # before/after comparison depends on, and reporting `run_id: run_001`
+        # for a solve of run_002's deck.
+        deck_run_id = _run_id_from_deck_path(input_deck_path_str)
+        if requested_run_id and deck_run_id and requested_run_id != deck_run_id:
+            return _with_run_solver_receipt({
```

Ruff (0.16.3)

[error] Undefined name `_with_run_solver_receipt`

(F821)